community/kernel/src/main/java/org/neo4j/kernel/impl/index/schema/SpatialIndexReader.java

```diff
@@ -0,0 +1,161 @@
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index.schema;
+
+import java.io.IOException;
+
+import org.neo4j.collection.primitive.PrimitiveLongResourceIterator;
+import org.neo4j.graphdb.Resource;
+import org.neo4j.helpers.collection.Iterators;
+import org.neo4j.internal.kernel.api.IndexOrder;
+import org.neo4j.internal.kernel.api.IndexQuery;
+import org.neo4j.internal.kernel.api.IndexQuery.ExistsPredicate;
+import org.neo4j.kernel.api.schema.index.SchemaIndexDescriptor;
+import org.neo4j.kernel.impl.index.schema.fusion.BridgingIndexProgressor;
+import org.neo4j.kernel.impl.index.schema.fusion.FusionIndexSampler;
+import org.neo4j.storageengine.api.schema.IndexProgressor;
+import org.neo4j.storageengine.api.schema.IndexReader;
+import org.neo4j.storageengine.api.schema.IndexSampler;
+import org.neo4j.values.storable.CoordinateReferenceSystem;
+import org.neo4j.values.storable.PointValue;
+import org.neo4j.values.storable.Value;
+
+import static org.neo4j.kernel.impl.index.schema.fusion.FusionIndexBase.forAll;
+
+class SpatialIndexReader extends SpatialIndexCache<SpatialIndexPartReader<SpatialSchemaKey,NativeSchemaValue>,IOException> implements IndexReader
+{
+    private final SchemaIndexDescriptor descriptor;
+
+    SpatialIndexReader( SchemaIndexDescriptor descriptor, SpatialIndexAccessor accessor )
+    {
+        super( new PartFactory( accessor ) );
+        this.descriptor = descriptor;
+    }
+
+    @Override
+    public void close()
+    {
+        forAll( Resource::close, this );
+    }
+
+    @Override
+    public long countIndexedNodes( long nodeId, Value... propertyValues )
+    {
+        NativeSchemaIndexReader<SpatialSchemaKey,NativeSchemaValue> partReader =
+                uncheckedSelect( ((PointValue) propertyValues[0]).getCoordinateReferenceSystem() );
+        return partReader == null ? 0L : partReader.countIndexedNodes( nodeId, propertyValues );
+    }
+
+    @Override
+    public IndexSampler createSampler()
+    {
+        return new FusionIndexSampler( Iterators.stream( iterator() ).map( IndexReader::createSampler ).toArray( IndexSampler[]::new ) );
+    }
+
+    @Override
+    public PrimitiveLongResourceIterator query( IndexQuery... predicates )
```

Comment:
Both the old implementation and the temporal version have a special case for `predicates[0] instanceof ExisitsPredicate` here. Was it obsolete due to the special handling of the same case in the `query` method below or why was it removed?